Gain performance by doing less.

heller opened the pull request:
Don't try to screenshot a domain permutation if it isn't registered.
Omit the calls to gendom_stop for a substantial performance gain.

f8al commented:
approved

f8al commented:
this adds performance successfully to the screenshot section, but the whois section is still trying to query nonexistent domains, or is saying valid domains are nonexistent but I wonder if this isn't just being broken by the damned whois library again?
<img width="1680" alt="image" src="https://github.com/f8al/DNSrazzle/assets/7221183/ef67f9bf-7208-436e-b116-80e721fd679e">

f8al commented:
can confirm, by pinning the old version of domainwhois this resolves but creates a new issue

heller commented:
Please include the command you ran when posting these kinds of reports so that I can attempt to reproduce it in the same way!

heller commented:
FWIW I only got one WHOIS failure when checking foo.com. (using `whoisdomain 1.20230720.2`.) Ditch the VPN again maybe?

```
(dnsrazzle) matt@0247 dnsrazzle % python -B DNSrazzle.py -d foo.com --browser firefox
 ______  __    _ _______ ______   _______ _______ _______ ___     _______
 |      ||  |  | |       |    _ | |   _   |       |       |   |   |       |
 |  _    |   |_| |  _____|   | || |  |_|  |____   |____   |   |   |    ___|
 | | |   |       | |_____|   |_||_|       |____|  |____|  |   |   |   |___ 
 | |_|   |  _    |_____  |    __  |       | ______| ______|   |___|    ___|
 |       | | |   |_____| |   |  | |   _   | |_____| |_____|       |   |___ 
 |______||_|  |__|_______|___|  |_|__| |__|_______|_______|_______|_______|

Version 1.5.3 by SecurityShrimp
[*] Saving records to output folder /Users/matt/Code/oss/dnsrazzle
Generating possible domain name impersonations… |################################| 1/1
Running DNS lookup of possible domain permutations for foo.com… |################################| 183/183
Running WHOIS queries on discovered domains for foo.com… |###########################     | 158/184[-] Failed to run WHOIS query for fkoo.com
[-] whois: whois.paycenter.com.cn: nodename nor servname provided, or not known: Invalid argument
% IANA WHOIS server
% for more information on IANA, visit http://www.iana.org
% This query returned 1 object

refer:        whois.verisign-grs.com

domain:       COM

organisation: VeriSign Global Registry Services
address:      12061 Bluemont Way
address:      Reston VA 20190
address:      United States of America (the)

contact:      administrative
name:         Registry Customer Service
organisation: VeriSign Global Registry Services
address:      12061 Bluemont Way
address:      Reston VA 20190
address:      United States of America (the)
phone:        +1 703 925-6999
fax-no:       +1 703 948 3978
e-mail:       info@verisign-grs.com

contact:      technical
name:         Registry Customer Service
organisation: VeriSign Global Registry Services
address:      12061 Bluemont Way
address:      Reston VA 20190
address:      United States of America (the)
phone:        +1 703 925-6999
fax-no:       +1 703 948 3978
e-mail:       info@verisign-grs.com

nserver:      A.GTLD-SERVERS.NET 192.5.6.30 2001:503:a83e:0:0:0:2:30
nserver:      B.GTLD-SERVERS.NET 192.33.14.30 2001:503:231d:0:0:0:2:30
nserver:      C.GTLD-SERVERS.NET 192.26.92.30 2001:503:83eb:0:0:0:0:30
nserver:      D.GTLD-SERVERS.NET 192.31.80.30 2001:500:856e:0:0:0:0:30
nserver:      E.GTLD-SERVERS.NET 192.12.94.30 2001:502:1ca1:0:0:0:0:30
nserver:      F.GTLD-SERVERS.NET 192.35.51.30 2001:503:d414:0:0:0:0:30
nserver:      G.GTLD-SERVERS.NET 192.42.93.30 2001:503:eea3:0:0:0:0:30
nserver:      H.GTLD-SERVERS.NET 192.54.112.30 2001:502:8cc:0:0:0:0:30
nserver:      I.GTLD-SERVERS.NET 192.43.172.30 2001:503:39c1:0:0:0:0:30
nserver:      J.GTLD-SERVERS.NET 192.48.79.30 2001:502:7094:0:0:0:0:30
nserver:      K.GTLD-SERVERS.NET 192.52.178.30 2001:503:d2d:0:0:0:0:30
nserver:      L.GTLD-SERVERS.NET 192.41.162.30 2001:500:d937:0:0:0:0:30
nserver:      M.GTLD-SERVERS.NET 192.55.83.30 2001:501:b1f9:0:0:0:0:30
ds-rdata:     30909 8 2 e2d3c916f6deeac73294e8268fb5885044a833fc5459588f4a9184cfc41a5766

whois:        whois.verisign-grs.com

status:       ACTIVE
remarks:      Registration information: http://www.verisigninc.com

created:      1985-01-01
changed:      2019-08-14
source:       IANA

# whois.verisign-grs.com

   Domain Name: FKOO.COM
   Registry Domain ID: 140661812_DOMAIN_COM-VRSN
   Registrar WHOIS Server: whois.paycenter.com.cn
   Registrar URL: http://www.xinnet.com
   Updated Date: 2020-12-09T04:14:37Z
   Creation Date: 2005-01-27T06:06:51Z
   Registry Expiry Date: 2025-01-27T06:06:51Z
   Registrar: Xin Net Technology Corporation
   Registrar IANA ID: 120
   Registrar Abuse Contact Email: supervision@xinnet.com
   Registrar Abuse Contact Phone: +86.4008182233
   Domain Status: ok https://icann.org/epp#ok
   Name Server: NS11.XINCACHE.COM
   Name Server: NS12.XINCACHE.COM
   DNSSEC: unsigned
   URL of the ICANN Whois Inaccuracy Complaint Form: https://www.icann.org/wicf/
>>> Last update of whois database: 2023-09-12T17:20:42Z <<<

# whois.paycenter.com.cn


Running WHOIS queries on discovered domains for foo.com… |################################| 184/184
[*] Processing domain information
[*] Domain data written to /Users/matt/Code/oss/dnsrazzle/discovered-domains.csv
[*] Collecting and analyzing web screenshots
<snip>
```
